src/test/java/org/sagebionetworks/bridge/services/AssessmentServiceTest.java

```diff
@@ -0,0 +1,1051 @@
+        
+        when(mockDao.getAssessment(APP_ID_VALUE, assessment.getGuid()))
+            .thenReturn(Optional.of(existing));
+        
+        service.updateAssessment(APP_ID_VALUE, assessment);
+    }
+    
+    @Test
+    public void updateAssessmentScrubsMarkup() {
+        when(mockSubstudyService.getSubstudy(APP_AS_STUDY_ID, OWNER_ID, false)).thenReturn(mockSubstudy);
+
+        Assessment assessment = AssessmentTest.createAssessment();
+        assessment.setDeleted(false);
+        addMarkupToSensitiveFields(assessment);
+
+        when(mockDao.getAssessment(eq(APP_ID_VALUE), any()))
+            .thenReturn(Optional.of(assessment));
+        
+        service.updateAssessment(APP_ID_VALUE, assessment);
+        
+        assertMarkupRemoved(assessment);
+    }
+   
+    @Test
+    public void updateSharedAssessment() {
+        BridgeUtils.setRequestContext(new RequestContext.Builder()
+                .withCallerStudyId(APP_AS_STUDY_ID)
+                .withCallerSubstudies(ImmutableSet.of(OWNER_ID)).build());
+        
+        when(mockSubstudyService.getSubstudy(APP_AS_STUDY_ID, OWNER_ID, false))
+            .thenReturn(Substudy.create());
+        
+        Assessment existing = AssessmentTest.createAssessment();
+        existing.setOriginGuid("unusualGuid");
+        existing.setDeleted(false);
+        existing.setOwnerId(APP_ID_VALUE + ":" + OWNER_ID);
+        when(mockDao.getAssessment(SHARED_STUDY_ID_STRING, GUID))
+            .thenReturn(Optional.of(existing));
+        
+        Assessment assessment = AssessmentTest.createAssessment();
+        service.updateSharedAssessment(APP_ID_VALUE, assessment);
+        
+        verify(mockDao).saveAssessment(SHARED_STUDY_ID_STRING, assessment);
+        
+        assertEquals(assessment.getIdentifier(), IDENTIFIER);
+        assertEquals(APP_ID_VALUE + ":" + OWNER_ID, assessment.getOwnerId());
+        assertEquals(assessment.getOriginGuid(), "unusualGuid");
+        assertEquals(assessment.getCreatedOn(), CREATED_ON);
+        assertEquals(assessment.getModifiedOn(), MODIFIED_ON);
+    }
+    
+    @Test(expectedExceptions = UnauthorizedException.class)
+    public void updateSharedAssessmentUnauthorizedApp() {
+        BridgeUtils.setRequestContext(new RequestContext.Builder()
+                .withCallerStudyId(APP_AS_STUDY_ID)
+                .withCallerSubstudies(ImmutableSet.of(OWNER_ID)).build());
+        
+        Assessment existing = AssessmentTest.createAssessment();
+        existing.setDeleted(false);
+        existing.setOwnerId("wrong-app:" + OWNER_ID);
+        when(mockDao.getAssessment(SHARED_STUDY_ID_STRING, GUID))
+            .thenReturn(Optional.of(existing));
+        
+        Assessment assessment = AssessmentTest.createAssessment();
+        service.updateSharedAssessment(APP_ID_VALUE, assessment);
+    }
+    
+    @Test(expectedExceptions = UnauthorizedException.class)
+    public void updateSharedAssessmentUnauthorizedOrg() {
```

Comment:
What about users without orgs?

Reply:
Added a test that they can edit shared assessments.